addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1266,6 +1270,88 @@ options.registry.parallax = options.Class.extend({
     onMove: function () {
         this._refreshPublicWidgets();
     },
+    /**
+     * @override
+     */
+    updateUI: async function () {
+        await this._super(...arguments);
+        // Remove parallax if no background-image.
+        // Example: click on the remove background-image button.
+        if (this.isParallax) {
+            let $el = this.$target.find('> .s_parallax_bg');
+            const hasBackgroundImage = $el.css('background-image') !== 'none';
+            if (!hasBackgroundImage) {
+                this._removeParallax();
+                this.$target.removeClass('oe_img_bg');
+            }
+        }
```

Comment:
Again: methods updating the UI **must not** modify the DOM. But I agree there is no clean way to do that for now :/ Maybe you could do something like the code of the background image being removed when a background color is picked. Or find a better way, snippet_option_update could ask all other options if they need something to be changed instead of just asking them to update their UI...